Mathlib/Algebra/BigOperators/Fin.lean

```diff
@@ -262,12 +262,71 @@ theorem inv_partialProd_mul_eq_contractNth {G : Type*} [Group G] (g : Fin (n + 1
 
 end PartialProd
 
+section DivModSum
+
+/-- Given a tuple of natural numbers `n : Fin m → ℕ` and a natural number `k`, return the first
+index `i` such that `k` is smaller than `∑ j < i, n j`, and `none` otherwise.
+
+This is defined en-route to `Fin.divSum`, which is the dependent version of `Fin.divNat`.
+-/
+def divSum? {m : ℕ} (n : Fin m → ℕ) (k : ℕ) : Option (Fin m) :=
+  find (fun i => k < ∑ j : Fin i.val.succ, n (castLE i.isLt j))
```

Comment:
Possibly better as
```suggestion
  find (fun i => k < ∑ j ∈ Finset.Iic i, n j)
```
which avoids the `.succ` and casting

Reply:
Yeah, this is certainly possible and looks nicer. I'd probably want to propagate this change to `finPiFinEquiv` as well (which a quick search tells me isn't used anywhere else). It also means that I'll need to add more lemmas about sum over `Finset.Iic`.

If all of this sounds good to you, then I'll make the change.

Reply:
Let's check @YaelDillies agrees too

Reply:
Yes, makes sense to me